### PR TITLE
[KS-490] Support per-step timeout overrides in the Engine

### PR DESCRIPTION
**Pull request by bolekk**

### CODE REVIEW

File: core/services/workflows/engine.go [hunk 1]
```diff
@@ -27,7 +27,11 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/core/services/workflows/store"
 )
 
-const fifteenMinutesMs = 15 * 60 * 1000
+const (
+	fifteenMinutesMs             = 15 * 60 * 1000
```
Comment (krehermann) [inline core/services/workflows/engine.go:31]: use time.Duration to avoid an ambiguity?

**Review by krehermann — APPROVED** [review]

File: core/services/workflows/engine.go (hunk 1 above)
Comment (cedric-cordenier) [reply core/services/workflows/engine.go:31]: Yeah let's please use time.Duration wherever we can :)

File: core/services/workflows/engine.go [hunk 2]
```diff
@@ -934,8 +949,11 @@ func (e *Engine) executeStep(ctx context.Context, lggr logger.Logger, msg stepRe
 		},
 	}
 
-	e.metrics.incrementCapabilityInvocationCounter(ctx)
-	output, err := step.capability.Execute(ctx, tr)
```
Comment (patrickhuie19) [inline core/services/workflows/engine.go:938]: Execute is syncronous, so this should not be a problem?

**Review by cedric-cordenier — APPROVED** [review]

File: core/services/workflows/engine.go (hunk 1 above)
Comment (patrickhuie19) [reply core/services/workflows/engine.go:31]: The tradeoff is you cant use `const` as the time.Duration methods are runtime only

File: core/services/workflows/engine.go [hunk 3]
```diff
@@ -27,7 +27,11 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/core/services/workflows/store"
 )
 
-const fifteenMinutesMs = 15 * 60 * 1000
+const (
+	fifteenMinutesMs             = 15 * 60 * 1000
+	reservedFieldNameStepTimeout = "cre_step_timeout"
+	maxStepTimeoutOverrideSec    = 10 * 60 // 10 minutes
+)
```
Comment (patrickhuie19) [inline core/services/workflows/engine.go:34]:
```suggestion
var (
	fifteenMinutesMs             = 15 * time.Minute.Milliseconds()
	reservedFieldNameStepTimeout = "cre_step_timeout"
	maxStepTimeoutOverrideSec    = 10 * time.Minute.Seconds()
)
```

File: core/services/workflows/engine.go [hunk 4]
```diff
@@ -919,6 +920,20 @@ func (e *Engine) executeStep(ctx context.Context, lggr logger.Logger, msg stepRe
 	if err != nil {
 		return nil, nil, err
 	}
+	stepTimeoutDuration := e.stepTimeoutDuration
+	if timeoutOverride, ok := config.Underlying[reservedFieldNameStepTimeout]; ok {
```
Comment (patrickhuie19) [inline core/services/workflows/engine.go:924]: is this config set per capability? Checking to make sure its not set at the workflow level

Comment (bolekk) [reply core/services/workflows/engine.go:924]: Yes, see example in the test file

File: core/services/workflows/engine.go (hunk 2 above)
Comment (bolekk) [reply core/services/workflows/engine.go:938]: what do you mean?

**Review by patrickhuie19 — APPROVED** [review]

**Pull request merged**